pkg/cmd/kafka/create/data.go

```diff
@@ -0,0 +1,149 @@
+package create
+
+import (
+	"github.com/redhat-developer/app-services-cli/pkg/shared/accountmgmtutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+)
+
+// Types we use on backend to map AMS Quotas
+type CloudProviderId = string
+
+// Additional types that are used in backend to match
+// AMS Quota instance types
+const (
+	// Matches QuotaTrialType = trial
+	DeveloperType CloudProviderId = "developer"
+
+	// Matches QuotaStandardType = standard
+	StandardType CloudProviderId = "standard"
+
+	// Deprecated by DeveloperType
+	TrialType CloudProviderId = "eval"
+)
+
+// mapAmsTypeToBackendType - Cloud providers API is not using AMS types but some other values (CloudProviderValues)
+func mapAmsTypeToBackendType(amsType *accountmgmtutil.QuotaSpec) CloudProviderId {
+	switch amsType.Name {
+	case accountmgmtutil.QuotaStandardType:
+		return StandardType
+	case accountmgmtutil.QuotaTrialType:
+		return DeveloperType
+	default:
+		return DeveloperType
+	}
+}
+
+// return list of the valid instance sizes for the specifed region and ams instance types
+func GetValidKafkaSizes(f *factory.Factory,
+	providerID string, regionId string, amsType accountmgmtutil.QuotaSpec) ([]string, error) {
+
+	conn, err := f.Connection(connection.DefaultConfigSkipMasAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	validSizes := []string{}
+
+	instanceTypes, _, err := conn.API().
+		KafkaMgmt().
+		GetInstanceTypesByCloudProviderAndRegion(f.Context, providerID, regionId).
+		Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	desiredInstanceType := mapAmsTypeToBackendType(&amsType)
+
+	// Temporary workaround to be removed
+	if desiredInstanceType == DeveloperType {
+		for _, instanceType := range instanceTypes.GetInstanceTypes() {
+			if desiredInstanceType == instanceType.GetId() {
+				for _, instanceSize := range instanceType.GetSizes() {
+					validSizes = append(validSizes, instanceSize.GetId())
+				}
+			}
+		}
+	} else {
+		for _, instanceType := range instanceTypes.GetInstanceTypes() {
+			if desiredInstanceType == instanceType.GetId() {
+				for _, instanceSize := range instanceType.GetSizes() {
+					if instanceSize.GetQuotaConsumed() <= int32(amsType.Quota) {
```

Comment:
Ok, I understand now why you're relying on `quota_consumed` being `0` for developer instances.

Reply:
Ah.. I forgot to changed it. In chat you have suggesed using different value. Does it matter?

Reply:
I meant should be using CapacityConsumed or Quota consumed to filter this?

Reply:
Overal I do not think we should have that much logic around instance types in the client. Limitation right now is that AMS and Backend diverge a lot so we would need to have those values in generic form to build generic client. 

I would even say that if we cannot reduce business logic we can pull out some validation logic you have on backend and reuse it as source of truth for SDK.